datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -1695,6 +1709,35 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn parquet_page_index_exec_metrics() {
+        let c1: ArrayRef = Arc::new(Int32Array::from(vec![Some(1), None, Some(2)]));
+        let c2: ArrayRef = Arc::new(Int32Array::from(vec![Some(3), Some(4), Some(5)]));
+        let batch1 = create_batch(vec![("int", c1.clone())]);
+        let batch2 = create_batch(vec![("int", c2.clone())]);
+
+        let filter = col("int").eq(lit(4_i32));
+
+        let rt =
+            round_trip(vec![batch1, batch2], None, None, Some(filter), false, true).await;
+
+        let metrics = rt.parquet_exec.metrics().unwrap();
+
+        // todo fix this  https://github.com/apache/arrow-rs/issues/2941 release change to row limit.
```

Comment:
https://github.com/apache/arrow-rs/pull/2942/files#r1013838557
I think theres is a bug in `should_add_data_page `  `self.encoder.num_values() ` always zero.
So no matter how to set  `data_pagesize_limit ` and `write_batch_size` always return 1 page in on colunn chunk.
🤔 I think someone metion this before.

Reply:
But from the real world file, this metic works fine😂